sdk/src/main/java/com/uid2/UID2Client.kt

```diff
@@ -20,34 +26,105 @@ import java.net.URL
  */
 internal class UID2Client(
     private val apiUrl: String,
+    private val apiPublicKey: String?,
+    private val apiSubscriptionId: String?,
     private val session: NetworkSession,
+    private val packageName: String,
+    private val dataEnvelope: DataEnvelope = DataEnvelope(),
+    private val timeUtils: TimeUtils = TimeUtils(),
+    private val keyUtils: KeyUtils = KeyUtils(),
     private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO,
+    private val paramsFactory: (Map<String, String>) -> String = { params ->
+        JSONObject().apply {
+            params.forEach { (key, value) -> put(key, value) }
```

Comment:
nit: up to you, you could use the JSONObject that takes a map
```suggestion
        JSONObject(params)
```

Reply:
Good point, thanks